test/e2e/framework/internal/unittests/testapiserver.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unittests
+
+import (
+	"path"
+	"testing"
+
+	"github.com/google/uuid"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+	etcdserver "k8s.io/apiserver/pkg/storage/etcd3/testserver"
+	"k8s.io/apiserver/pkg/storage/storagebackend"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/cert"
+	kubeapiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+)
+
+type TestAPIServer struct {
+	ClientSet      clientset.Interface
+	KubeConfigFile string
+}
+
+func StartAPITestServer(t *testing.T) TestAPIServer {
```

Comment:
I am undecided whether this helper should be here or somewhere else, like `test/utils/apiserver`. Therefore I also haven't written any documentation for it.

Any thoughts?

Reply:
I moved it to `test/utils/apiserver`. That also made `verify-import-boss.sh` happy, although I don't fully understand why depending on the same packages indirectly is fine when depending on them directly is not.

It doesn't matter for `test/e2e/framework` because the goal no longer is to move it to staging.